src/vfs.c

```diff
@@ -2598,7 +2603,8 @@ static int vfsDatabaseRestore(struct vfsDatabase *d,
 			goto oom_after_pages_alloc;
 		}
 		pages[i] = page;
-		memcpy(page, &data[i * page_size], page_size);
+		offset = (size_t)i * (size_t)page_size;
```

Comment:
This indexes into a data buffer of maximum size `SIZE_MAX` and we already check here https://github.com/canonical/dqlite/blob/5d00a628786a4c078e66e4c8dec9e8232e440d9f/src/fsm.c#L464 if the buffer is smaller than `SIZE_MAX`, so this should not overflow.